definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx

```diff
@@ -0,0 +1,46 @@
+/*
+* Copyright 2023 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+config {
+  type: "view",
+  schema: functions.productSchema("ga4"),
+  description: "Latest event per user within the last 72 hours window",
+  tags: ['ga4']
+}
+
+js {
+  const columns = [
+    'user_pseudo_id',
+    'user_id',
+    'event_id',
+    'event_name',
+    'event_type',
+    'event_timestamp'
+  ]
+}
+
+SELECT
+${columns}
+FROM (
+  SELECT
+    ${columns},
+    ROW_NUMBER() OVER (PARTITION BY user_pseudo_id ORDER BY event_timestamp DESC) AS rn
```

Comment:
Is user_pseudo_id truly globally unique? I don't think so, but maybe mistaken. If not - need to have another column(s) in the PARTITION BY clause.

Reply:
added group by user_id together with user_pseudo_id which is the only two user identifier in the data.